legate/driver/args.py

```diff
@@ -46,50 +47,153 @@
 __all__ = ("parser",)
 
 
-def detect_multi_node_defaults() -> tuple[dict[str, Any], dict[str, Any]]:
-    from os import getenv
+def _get_ompi_config() -> tuple[int, int] | None:
+    if not (ranks_env := getenv("OMPI_COMM_WORLD_SIZE")):
+        return None
+
+    if not (ranks_per_node_env := getenv("OMPI_COMM_WORLD_LOCAL_SIZE")):
+        return None
+
+    try:
+        ranks, ranks_per_node = int(ranks_env), int(ranks_per_node_env)
+    except ValueError:
+        raise ValueError(
+            "Expected OMPI_COMM_WORLD_SIZE and OMPI_COMM_WORLD_LOCAL_SIZE to "
+            f"be integers, got OMPI_COMM_WORLD_SIZE={ranks_env} and "
+            f"OMPI_COMM_WORLD_LOCAL_SIZE={ranks_per_node_env}"
+        )
+
+    if ranks % ranks_per_node != 0:
+        raise ValueError(
+            "Detected incompatible ranks and ranks-per-node from "
+            f"OMPI_COMM_WORLD_SIZE={ranks} and "
+            f"OMPI_COMM_WORLD_LOCAL_SIZE={ranks_per_node}"
+        )
+
+    return ranks // ranks_per_node, ranks_per_node
+
+
+def _get_mv2_config() -> tuple[int, int] | None:
+    if not (ranks_env := getenv("MV2_COMM_WORLD_SIZE")):
+        return None
+
+    if not (ranks_per_node_env := getenv("MV2_COMM_WORLD_LOCAL_SIZE")):
+        return None
+
+    try:
+        ranks, ranks_per_node = int(ranks_env), int(ranks_per_node_env)
+    except ValueError:
+        raise ValueError(
+            "Expected MV2_COMM_WORLD_SIZE and MV2_COMM_WORLD_LOCAL_SIZE to "
+            f"be integers, got MV2_COMM_WORLD_SIZE={ranks_env} and "
+            f"MV2_COMM_WORLD_LOCAL_SIZE={ranks_per_node_env}"
+        )
+
+    if ranks % ranks_per_node != 0:
+        raise ValueError(
+            "Detected incompatible ranks and ranks-per-node from "
+            f"MV2_COMM_WORLD_SIZE={ranks} and "
+            f"MV2_COMM_WORLD_LOCAL_SIZE={ranks_per_node}"
+        )
+
+    return ranks // ranks_per_node, ranks_per_node
+
+
+def _get_slurm_config() -> tuple[int, int] | None:
+    if not (nodes_env := getenv("SLURM_JOB_NUM_NODES")):
+        return None
+
+    nprocs_env = getenv("SLURM_NPROCS")
+    ntasks_env = getenv("SLURM_NTASKS")
+    tasks_per_node_env = getenv("SLURM_TASKS_PER_NODE")
+
+    # at least one of these needs to be set
+    if (nprocs_env, ntasks_env, tasks_per_node_env) == (None, None, None):
+        return None
+
+    # use SLURM_TASKS_PER_NODE if it is given
+    if tasks_per_node_env is not None:
+        if ntasks_env is not None:
+            raise ValueError(
+                "Ambiguous congfiguration, both SLURM_TASKS_PER_NODE and "
+                "SLURM_NTASKS are set."
+            )
+        if nprocs_env is not None:
+            raise ValueError(
+                "Ambiguous congfiguration, both SLURM_TASKS_PER_NODE and "
+                "SLURM_NPROCS are set."
+            )
```

Comment:
I think it's quite possible that all are defined at the same time, so I think we need to remove this check.

Reply:
https://github.com/nv-legate/legate.core/pull/917/commits/20bfa5505fb4c3295acddee06465d2b66661e56d